packages/zipkin/src/option.js

```diff
@@ -2,44 +2,45 @@ const None = {
   get type() {
     return 'None';
   },
+  get present() {
+    return false;
+  },
   map() {
-    return None;
+    return this;
   },
```

Comment:
I think it's just tradeoffs, not necessarily better 😄.  Here are a couple pros and cons I considered--maybe there are others.  Ultimately, I'm cool with whichever you'd prefer.

I felt one pro of `this` is consistency: `this` can be used consistently in every None and Some method, whereas `None` can only be used in the None methods, and `this` must be used in the Some methods.  On the other hand, I think using `None` reads nicely and communicates intent slightly better.

Using `None` permits calling None's methods as functions (i.e. without a context/thisArg), but Some's methods must only be called as methods.  That difference seems odd, although my guess is that it's unlikely ever to trip anyone up in practice.

Reply:
It is great to know your thoughts on the tradeoffs. Please keep what you favor most.
I also think both are fine in this case, but if I implement it, I will slightly favor None because of the benefits that you described(rigid intent). ^__^